src/Spain/Mews.Fiscalizations.Spain/Communication/SoapClient.cs

```diff
@@ -47,15 +47,19 @@ internal SoapClient(Uri endpointUri, X509Certificate certificate, TimeSpan httpT
             {
                 var soapResponseBody = GetSoapBody(response);
                 var deserializedMessage = XmlSerializer.Deserialize<TOut>(soapResponseBody.OuterXml);
-                return new ResponseResult<TOut>(successResult: deserializedMessage);
+                return ResponseResult.Success(deserializedMessage);
             }
             catch (InvalidOperationException)
             {
                 var soapFault = XmlSerializer.Deserialize<SoapFaultResponse>(response);
-                return new ResponseResult<TOut>(errorResult: new ErrorResult(
-                    code: soapFault.Body.Fault.Faultcode,
-                    message: soapFault.Body.Fault.Faultstring
-                ));
+                try
+                {
+                    return ResponseResult.Error<TOut>(soapFault.Body.Fault.Code, soapFault.Body.Fault.Message);
+                }
+                catch (Exception e)
+                {
+                    throw new InvalidOperationException($"Fail to parse response {response}", e);
+                }
```

Comment:
IMO this is not needed, because if deserializing the response to SoapFaultResponse was successful, then we know we have these values as the server-side will always return them.

Reply:
I put this just to help future investigations. If there is a fault in xml deserialization there is not way to get the message. I believe there is a decent chance of having an issue since we do not really consume code-generated service based on wsdl, but created all the contracts manually.

Reply:
I would like to suggest to keep this at least at first time and remove afterwards.

Reply:
@KaliCZ I really still think this doesn't make any sense and it won't help

if `var soapFault = XmlSerializer.Deserialize<SoapFaultResponse>(response);` succeeds, we must get a valid SOAP fault exeption https://docs.microsoft.com/en-us/dotnet/api/system.runtime.serialization.formatters.soapfault?view=netframework-4.8

Reply:
I don't understand what this try-catch is acually catching. It seems to me that the only thing it can catch is a null pointer in the `SoapFaultResponse` body.

In that case, I wouldn't really handle the null with a try-catch. There's 2 options here:
- It can never be null. In that case this is useless code. 
- It actually can be null. Then we should handle it properly and not with a general try-catch.

So I would get rid of it. And if it happens that nulls are happening, we can handle them.